File: src/main/java/akostenko/aicars/race/cartest/AccelerationAndBreakingTest.java
package akostenko.aicars.race.cartest;

import akostenko.aicars.race.Driver;

public class AccelerationAndBreakingTest extends Driver {

    private final double targetSpeed; // m/s
    private double time = 0;
    private boolean speedReached;
    private boolean targetReached;

    public AccelerationAndBreakingTest(double targetSpeedKMH) {
        this.targetSpeed = targetSpeedKMH;
    }

    @Override
    public String getName() {
        return String.format("0-%d-0 kph", (int) targetSpeed) + (targetReached ? String.format(" %.3fs", time) : "");
    }

    @Override
    public double accelerating() {
        return !speedReached ? 1 : 0;
    }

    @Override
    public double breaking() {
        return speedReached ? 1 : 0;
    }

    @Override
    public double steering() {
        return 0;
    }

    @Override
    public void update(double dTime) {
        if (!targetReached) {
            speedReached = speedReached || getCar().speed().module() * 3.6 >= targetSpeed;
            targetReached = speedReached && getCar().speed().module() <= 0.1;
            time += dTime;
        }
    }
}

File: src/main/java/akostenko/aicars/race/cartest/AccelerationAndBreakingTest.kt
package akostenko.aicars.race.cartest

import akostenko.aicars.race.Driver

class AccelerationAndBreakingTest(private val targetSpeed: Double // m/s
) : Driver() {
    private var time = 0.0
    private var speedReached: Boolean = false
    private var targetReached: Boolean = false

    override val name: String
        get() = String.format("0-%d-0 kph", targetSpeed.toInt()) + if (targetReached) String.format(" %.3fs", time) else ""

    override fun accelerating(): Double {
        return (if (!speedReached) 1 else 0).toDouble()
    }

    override fun breaking(): Double {
        return (if (speedReached) 1 else 0).toDouble()
    }

    override fun steering(): Double {
        return 0.0
    }

    override fun update(dTime: Double) {
        if (!targetReached) {
            speedReached = speedReached || car.speed().module() * 3.6 >= targetSpeed
            targetReached = speedReached && car.speed().module() <= 0.1
            time += dTime
        }
    }
}
